src/main/java/com/minecolonies/coremod/colony/ColonyList.java

```diff
@@ -0,0 +1,271 @@
+package com.minecolonies.coremod.colony;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.minecolonies.coremod.util.Log;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+//TODO extend list/collection
+/**
+ * Data structure for storing colonies, optimized for performance.
+ *
+ * @author Colton
+ *
+ * @param <T> Type of IColony (Colony or ColonyView)
+ */
+public final class ColonyList<T extends IColony> implements Iterable<T>
+{
+    @VisibleForTesting
+    static final int INITIAL_SIZE = 16;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Explicitly declare the visibility for "INITIAL_SIZE". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2039)